api/src/main/java/com/michelin/ns4kafka/controllers/SchemaController.java

```diff
@@ -0,0 +1,179 @@
+        }
+
+        // Validate ownership
+        if (!this.schemaService.isNamespaceOwnerOfSubject(ns, schema.getMetadata().getName())) {
+            throw new ResourceValidationException(List.of("Invalid value " + schema.getMetadata().getName() +
+                    " for name: : namespace not OWNER of underlying topic"), schema.getKind(), schema.getMetadata().getName());
+        }
+
+        // Validate compatibility
+        List<String> validationErrors = this.schemaService.validateSchemaCompatibility(ns.getMetadata().getCluster(), schema);
+
+        if (!validationErrors.isEmpty()) {
+            throw new ResourceValidationException(validationErrors, schema.getKind(), schema.getMetadata().getName());
+        }
+
+        Optional<Schema> existingSchemaOptional = this.schemaService.getLatestSubject(ns, schema.getMetadata().getName());
+
+        ApplyStatus status = existingSchemaOptional.isPresent() ? ApplyStatus.changed : ApplyStatus.created;
+
+        if (dryrun) {
+            return this.formatHttpResponse(schema, status);
+        }
+
+        Optional<Schema> schemaOptional = this.schemaService.register(ns, schema);
+        status = schemaOptional.isPresent() ? status : ApplyStatus.unchanged;
+        Schema appliedSchema = schemaOptional.orElse(null);
+
+        return this.formatHttpResponse(appliedSchema, status);
+    }
+
+    /**
+     * Delete all schemas under the given subject
+     *
+     * @param namespace The current namespace
+     * @param subject The current subject to delete
+     * @param dryrun Run in dry mode or not
+     * @return A HTTP response
+     */
+    @Status(HttpStatus.NO_CONTENT)
+    @Delete("/{subject}")
+    public HttpResponse<Void> deleteSubject(String namespace, @PathVariable String subject,
+                                              @QueryValue(defaultValue = "false") boolean dryrun) {
+        Namespace retrievedNamespace = super.getNamespace(namespace);
+
+        if (!this.schemaService.isNamespaceOwnerOfSubject(retrievedNamespace, subject)) {
+            throw new ResourceValidationException(List.of("Invalid prefix " + subject +
+                    " : namespace not owner of this subject"), AccessControlEntry.ResourceType.SCHEMA.toString(), subject);
+        }
+
+        if (dryrun) {
+            return HttpResponse.noContent();
+        }
+
+        this.schemaService.deleteSubject(retrievedNamespace, subject);
+
+        return HttpResponse.noContent();
+    }
+
+    /**
+     * Update the compatibility of a subject
+     *
+     * @param namespace The namespace
+     * @param subject The subject to update
+     * @param compatibility The compatibility to apply
+     * @param dryrun Dry run mode
+     * @return The last updated schema
+     */
+    @Post("/{subject}/compatibility")
+    public HttpResponse<Optional<Schema>> compatibility(String namespace, @PathVariable String subject, @Valid @Body Map<String, Schema.Compatibility> compatibility,
```

Comment:
Simple body maps to JSON directly
```suggestion
    public HttpResponse<Optional<Schema>> compatibility(String namespace, @PathVariable String subject, @Body Schema.Compatibility compatibility,
```

Reply:
Mapping a simple String to Schema.Compatibility directly seems not working with JSON format